sunpy/visualization/animator/base.py

```diff
@@ -558,7 +558,7 @@ def plot_start_image(self):  # pragma: no cover
         raise NotImplementedError("Must be defined and used by subclasses of {0}.".format(self.__class__))
 
     @abc.abstractmethod
-    def update_plot(self, val, slider):  # pragma: no cover
+    def update_plot(self, val, im=None, slider):  # pragma: no cover
```

Comment:
Can you explain this for me?

Reply:
Running the ImageAnimatorWCS example from docs, you get :
``` File "/mnt/h/Documents/Projects/sunpy-dev/sunpy/visualization/animator/image.py", line 119, in update_plot
    super().update_plot(val, im, slider)
TypeError: update_plot() takes 3 positional arguments but 4 were given
Aborted (core dumped)
```
error when playing the animation. Apparently, `update_plot` in base takes in two inputs, while three were given in ImageAnimatorWCS class.
(Perhaps there's a better fix?)

What does `# pragma: no cover` mean?

Reply:
`# pragma: no cover` lets any code checker to ignore this line. You should remove it, I am not sure why it is there. 

But `im` is not used in that function as I can see? I'm not sure why it would need to be passed in.

Reply:
I would at least remove the `=None`.

Reply:
Traceback:
```Traceback (most recent call last):
  File "/home/akram/miniconda3/envs/sunpy-dev/lib/python3.7/site-packages/matplotlib/backend_bases.py", line 1238, in _on_timer
    ret = func(*args, **kwargs)
  File "/mnt/h/Documents/Projects/sunpy-dev/sunpy/visualization/animator/base.py", line 363, in _step
    s.set_val(s.val+1)
  File "/home/akram/miniconda3/envs/sunpy-dev/lib/python3.7/site-packages/matplotlib/widgets.py", line 438, in set_val
    func(val)
  File "/mnt/h/Documents/Projects/sunpy-dev/sunpy/visualization/animator/base.py", line 332, in _slider_changed
    self.slider_functions[slider.slider_ind](val, self.im, slider)
  File "/mnt/h/Documents/Projects/sunpy-dev/sunpy/visualization/animator/image.py", line 220, in update_plot
    super().update_plot(val, im, slider)
  File "/mnt/h/Documents/Projects/sunpy-dev/sunpy/visualization/animator/image.py", line 119, in update_plot
    super().update_plot(val, im, slider)
TypeError: update_plot() takes 3 positional arguments but 4 were given
Aborted (core dumped)
```

So yeah, lines 119 and 220 of image.py call `update_plot` from base ArrayAnimator.

Reply:
As for `im`'s usage, you are right, it is not needed. So well I'll just delete it from 119 & 220 of image.(?)

Reply:
I tried it that way and it doesn't seem to work.

Reply:
> I would at least remove the `=None`. If the other two values are not set, I think that `im` should be the same.

Well, as for that, it needs to be checked if there are any other calls where setting `im` could cause a problem.

Reply:
You could remove `im`. You have to check the functions correctly. Since I think it goes base -> image -> imageWCS.

Reply:
There is a call as such(super) in `line`, which just calls the two values.

Reply:
What about those lines?

Reply:
They call `update_plot` with all the three arguments.

Reply:
Can you also change line 41 to 
```
widget instance passed to them, e.g., ``update_slider(val, im, slider)``
```

Reply:
Sure.

Reply:
Well then we should add `im` to that function and leave it as is.

Reply:
I feel I got a solution. I'll push...